packages/ethers-5/src/client.ts

```diff
@@ -0,0 +1,788 @@
+    }
+  }
+
+  private async syncProfile(address: Address) {
+    const ensProvider = new ethers.providers.InfuraProvider('mainnet')
+    const name = await ensProvider.lookupAddress(address)
+    const avatar = await ensProvider.getAvatar(address)
+    if (name) {
+      this.setProfileName(name)
+    }
+    if (avatar) {
+      this.setProfileImage(avatar)
+    }
+  }
+
+  private async syncBalance(address: Address) {
+    const chainId = ProviderController.state.chainId
+    if (chainId) {
+      const networkRpcUrl = NetworkRPCUrls[chainId]
+      const networkName = NetworkNames[chainId]
+      const networkCurreny = networkCurrenySymbols[chainId]
+
+      if (networkRpcUrl && networkName && networkCurreny) {
+        const JsonRpcProvider = new ethers.providers.JsonRpcProvider(NetworkRPCUrls[chainId], {
+          chainId,
+          name: networkName
+        })
+        if (JsonRpcProvider) {
+          const balance = await JsonRpcProvider.getBalance(address)
+          const formattedBalance = utils.formatEther(balance)
+          this.setBalance(formattedBalance, networkCurreny)
+        }
+      }
+    }
+  }
+
+  private async switchNetwork(chainId: number) {
+    const provider = ProviderController.state.provider
+    const providerType = ProviderController.state.providerType
+
+    if (providerType === WALLET_CONNECT_CONNECTOR_ID) {
+      const walletConnectProvider = provider?.provider as EthereumProvider
+      if (walletConnectProvider) {
+        try {
+          await walletConnectProvider.request({
+            method: 'wallet_switchEthereumChain',
+            params: [{ chainId: `0x${chainId.toString(16)}` }]
+          })
+          ProviderController.setChainId(chainId)
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (switchError: any) {
+          if (switchError.code === 4902 || switchError.code === 5000) {
+            await walletConnectProvider.request({
+              method: 'wallet_addEthereumChain',
+              params: [
+                {
+                  chainId: `0x${chainId.toString(16)}`,
+                  rpcUrls: [NetworkRPCUrls[chainId]],
+                  chainName: NetworkNames[chainId],
+                  nativeCurrency: {
+                    name: networkCurrenySymbols[chainId],
+                    decimals: 18,
+                    symbol: networkCurrenySymbols[chainId]
+                  },
+                  blockExplorerUrls: [NetworkBlockExplorerUrls[chainId]],
+                  iconUrls: [NetworkImageIds[chainId]]
+                }
+              ]
+            })
+          } else {
+            throw new Error('Chain is not supported')
+          }
+        }
+      }
+    } else if (providerType === INJECTED_CONNECTOR_ID) {
+      const injectedProvider = provider
+      if (injectedProvider) {
+        try {
+          await injectedProvider.send('wallet_switchEthereumChain', [
+            { chainId: `0x${chainId.toString(16)}` }
```

Comment:
`0x${chainId.toString(16)}` bit is repeating a lot, could make it a utility. Same for other places that repeat (if any).